system.md

```diff
@@ -6,4 +6,87 @@ order: 4
 menu: contest
 ---
 
-To be announced.
\ No newline at end of file
+The system is based on Ubuntu 18.04 LTS.
+It will be possible to submit solutions using C99, C++17, Java 11, Python 2.7, and Python 3.6.
+Kotlin will also be available for company teams.
+
+## Documentation
+Documentation for C, C++, Java, Python 2, Python 3, and Kotlin will be available through [zeal](https://zealdocs.org/).
+
+## Compilation options
+(NOTE: these are subject to minor changes before the contest)
+For C and C++, the flag `-g` will be disabled on the judge systems.
+During the contest, the commands will be provided respectively as `mygcc`, `myg++`, `myjavac`, `myjava`, `mypython2` and `mypython3`.
+
+### C99
+```
+gcc -x c -Wall -O2 -std=gnu99 -static -pipe -o $DEST "$@" -lm
+```
+
+### C++17
+```
+g++ -x c++ -Wall -O2 -std=c++17 -static -pipe -o $DEST "$@"
+```
+
+### Java
+
+#### Compilation:
+````
+javac -encoding UTF-8 -d . "$@"
+````
+
+#### Runtime:
+````
+java -Xrs -XX:+UseSerialGC -Xss65536k -Xms1441792k -Xmx1441792k $MAINCLASS
+````
+
+### Python 2
+````
+pypy "$@"
+````
+
+### Python 3
+```
+python3 "$@"
+```
+
+## Available software
+NOTE: as the exact versions of installed software are not under our control, the actually available version might differ slightly.
+
+For each language, only the compiler and its standard libraries will be available on the judge systems!
```

Comment:
Maybe a note that there might be more libraries available on the contestant systems?